src/tool/xmeta/cpp/models/compilation_unit.cpp

```diff
@@ -0,0 +1,62 @@
+#include "compilation_unit.h"
+
+namespace xlang::xmeta
+{
+    compilation_error symbol_table::set_symbol(std::string_view symbol, type_category const& class_type)
```

Comment:
Compilation_error isn't clear here - are we talking errors that are raised during parsing or during a later validation step?

Reply:
They are suppose to be semantic errors. I will change the name to reflect that.